### PR TITLE
test: move accounting_tests and rpc_wallet_tests to wallet/test

### PR DESCRIPTION
**Pull request by laanwj**
Move the two other wallet tests to where they belong.

@jonasschnelli I'm surprised you didn't stumble on this one yet

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Yes! This should have been done long ago. Thanks for fixing.
utACK de39c95c2468cc643a5d5bed9e5dd8cea6a40747

**Comment by laanwj** [comment]
With this, I guess we also want a wallet-specific test fixture, to get rid of the `ENABLE_WALLET`s in `test_bitcoin.cpp`?

**Comment by laanwj** [comment]
Done. Latest commit removes all wallet-related setup from `test_bitcoin.cpp`. Wallet tests set up and tear down their own environment.

These references to the wallet remain in `src/test`, this probably signifies some code movement is still necessary:
```
multisig_tests.cpp:#include "wallet/wallet_ismine.h"
script_P2SH_tests.cpp:#include "wallet/wallet_ismine.h"
```

**Comment by sipa** [comment]
I believe the wallet_ismine code should be moved to script/sign or
script/standard.

File: src/wallet/test/wallet_test_fixture.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#include "wallet/db.h"
+#include "wallet/wallet.h"
+
+WalletTestingSetup::WalletTestingSetup(const std::string& chainName):
+    TestingSetup(chainName)
+{
+    bitdb.MakeMock();
+
+    bool fFirstRun;
+    pwalletMain = new CWallet("wallet.dat");
```
Comment (MarcoFalke) [inline src/wallet/test/wallet_test_fixture.cpp]: Could make sense to use "wallet_test.dat" here, to indicate that the file name is not hardcoded?

**Comment by laanwj** [comment]
@sipa What about keeping it separate and just moving, e.g. `script/ismine.cpp` and `script/ismine.h`?

File: src/wallet/test/wallet_test_fixture.cpp (hunk 1 above)
Comment (laanwj) [inline src/wallet/test/wallet_test_fixture.cpp]: Sounds good to me in principle (although I just moved this code here, I didn't touch it)

**Comment by laanwj** [comment]
> moving, e.g. script/ismine.cpp and script/ismine.h?

Done. All conditional wallet stuff is gone now from `src/test`.

**Comment by MarcoFalke** [comment]
Confirmed move only besides b30fb42

utACK b30fb42

**Pull request merged**